src/main/java/com/github/mc1arke/sonarqube/plugin/CommunityBranchPlugin.java

```diff
@@ -67,12 +73,106 @@ public void load(CoreExtension.Context context) {
                 PropertyDefinition.builder(CoreProperties.LONG_LIVED_BRANCHES_REGEX).onQualifiers(Qualifiers.PROJECT)
                         .category(CoreProperties.CATEGORY_GENERAL).subCategory(CoreProperties.SUBCATEGORY_BRANCHES)
-                        .defaultValue(CommunityBranchConfigurationLoader.DEFAULT_BRANCH_REGEX).build());
+                        .defaultValue(CommunityBranchConfigurationLoader.DEFAULT_BRANCH_REGEX).build(),
+
+                PropertyDefinition.builder("sonar.pullrequest.provider")
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Provider")
+                        .type(PropertyType.SINGLE_SELECT_LIST)
+                        .options("Github", "BitbucketServer", "BitbucketCloud").build(),
+
+                PropertyDefinition.builder(PullRequestBuildStatusDecorator.PULL_REQUEST_COMMENT_SUMMARY_ENABLED)
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Enable summary comment")
+                        .description("This enables the summary comment (if implemented).")
+                        .type(PropertyType.BOOLEAN)
+                        .defaultValue("true")
+                        .build(),
+
+                PropertyDefinition.builder(PullRequestBuildStatusDecorator.PULL_REQUEST_FILE_COMMENT_ENABLED)
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Enable file comment")
+                        .description("This enables commenting (if implemented).")
+                        .type(PropertyType.BOOLEAN)
+                        .defaultValue("true")
+                        .build(),
+
+                PropertyDefinition.builder(PullRequestBuildStatusDecorator.PULL_REQUEST_DELETE_COMMENTS_ENABLED)
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Enable deleting comments")
+                        .description("This cleans up the comments from previous runs (if implemented).")
+                        .type(PropertyType.BOOLEAN)
+                        .defaultValue("false")
+                        .build(),
+
+                PropertyDefinition.builder(BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_URL)
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(BITBUCKET_INTEGRATION_SUBCATEGORY_LABEL)
+                        .onQualifiers(Qualifiers.PROJECT)
```

Comment:
Should this be `Qualifier.PROJECT` or `Qualifier.APP`? Is `PROJECT` then the token needs set in every project in SonarQube, if `APP` then it should be set globally across all projects?